src/shims/foreign_items.rs

```diff
@@ -211,6 +214,89 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.machine.static_roots.push(ptr.alloc_id);
             }
 
+            // Obtains a Miri backtrace. See the README for details.
+            "miri_get_backtrace" => {
+                let tcx = this.tcx;
+                let mut data = Vec::new();
+                for frame in this.active_thread_stack().iter().rev() {
+                    data.push((frame.instance, frame.current_span().lo()));
+                }
+
+                let ptrs: Vec<_> = data.into_iter().map(|(instance, pos)| {
+                    let mut fn_ptr = this.memory.create_fn_alloc(FnVal::Instance(instance));
+                    fn_ptr.offset = Size::from_bytes(pos.0);
+                    Scalar::Ptr(fn_ptr)
+                }).collect();
+
+                let len = ptrs.len();
+
+                let ptr_ty = tcx.mk_ptr(TypeAndMut {
+                    ty: tcx.types.unit,
+                    mutbl: Mutability::Mut
+                });
+
+                let array_ty = tcx.mk_array(ptr_ty, ptrs.len().try_into().unwrap());
+                let array_ty_and_env = ParamEnv::empty().and(array_ty);
+
+                // Write pointers into array
+                let alloc = this.allocate(tcx.layout_of(array_ty_and_env).unwrap(), MiriMemoryKind::Rust.into());
+                for (i, ptr) in ptrs.into_iter().enumerate() {
+                    let place = this.mplace_index(alloc, i as u64)?;
+                    this.write_immediate_to_mplace(ptr.into(), place)?;
+                }
+
+                this.write_immediate(Immediate::new_slice(alloc.ptr.into(), len.try_into().unwrap(), this), dest)?;
+            }
+
+            // Resolves a Miri backtrace frame. See the README for details.
+            "miri_resolve_frame" => {
+                let tcx = this.tcx;
+                let &[ptr, flags] = check_arg_count(args)?;
+
+                let flags = this.read_scalar(flags)?.to_u64()?;
+                if flags != 0 {
+                    throw_ub_format!("Unknown `miri_resolve_frame` flags {}", flags);
```

Comment:
I think `throw_unsup_format!` would be more appropriate here. UB errors point to the reference page on UB, which makes no sense here.

Reply:
Also errors should start lower-case like in rustc. (Not all existing errors are consistent but we should do it for new ones at least.)